drivers/clock_control/clock_control_nrf.c

```diff
@@ -645,6 +645,12 @@ static int clk_init(const struct device *dev)
 		z_nrf_clock_calibration_init(data->mgr);
 	}
 
+#if defined(CONFIG_SOC_NRF5340_CPUAPP) && !defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
```

Comment:
This is not needed. The `SOC_CLOCK_128MHZ_NRF53X_APP` is anyway dependent on `SOC_NRF5340_CPUAPP` and there is no need to limit its availability to only secure code (the divider register in the CLOCK peripheral is accessible from both secure and non-secure code).